app/api/services/suppliers.py

```diff
@@ -373,3 +382,56 @@ def get_approved_suppliers(self):
         )
 
         return [r._asdict() for r in results]
+
+    def get_business_info_by_abn(self, email_address, abn):
+        apiKey = current_app.config['ABR_API_KEY']
+        includeHistoricalDetails = 'N'
+        abn = abn
+        link = 'https://abr.business.gov.au/abrxmlsearch/AbrXmlSearch.asmx/SearchByABNv201205?searchString='
+        url = link + abn + '&includeHistoricalDetails=' + includeHistoricalDetails + '&authenticationGuid=' + apiKey
+
+        try:
+            response = requests.get(url)
+            response.raise_for_status()
+            xmlText = response.content
+            root = ElementTree.fromstring(xmlText)
+
+        # Rasing Different exceptions
+        except ConnectionError as ex:
+            raise AbrError('Connection Error')
+
+        # Invalid HTTP Reponse
+        except HTTPError as ex:
+            raise AbrError('HTTP Error')
+
+        except ProxyError as ex:
+            raise AbrError('ProxyError')
+
+        except Timeout as ex:
+            raise AbrError('Timeout')
+
+        except SSLError as ex:
+            raise AbrError('SSLError')
+
+        # Any other expections
+        except Exception as ex:
+            raise AbrError('Failed exception raised')
+
+        # takes the first organisationName
+        search_xml_organisation_name = re.findall(r'<organisationName>(.*?)</organisationName>', xmlText)
+        organisation_name = search_xml_organisation_name[0]
+        # this only works for &, < and > but not ' and ""
+        organisation_name = saxutils.unescape(organisation_name)
+
+        # takes the first postcode
+        search_xml_postcode = re.findall(r'<postcode>(.*?)</postcode>', xmlText)
+        postcode = search_xml_postcode[0]
+
+        # takes the first state
+        search_xml_state = re.findall(r'<stateCode>(.*?)</stateCode>', xmlText)
+        state = search_xml_state[0]
+
+        # a dict to store these pre-filled info
+        business_info_abn_dict = {'organisation_name': organisation_name, 'postcode': postcode, 'state': state}
+        business_info_abn = json.dumps(business_info_abn_dict)
+        return business_info_abn
```

Comment:
Should we condense this down to the following?
```suggestion
        return json.dumps({
            'organisation_name': organisation_name,
            'postcode': postcode,
            'state': state
        })
```